### PR TITLE
[ART-8304] Enhance Dockerfiles to Utilize install.sh Script

### PR DESCRIPTION
**Pull request by MartinPl0**
Adjusted Dockerfile.base and Dockerfile.update to optimize the use of the install.sh script from art-tools. This change ensures proper execution of the script within the Docker build process, improving dependency management and aligning with our updated build strategy. The modifications include updating pip before the script execution and ensuring compatibility in the Docker environment.

### CODE REVIEW

**Review by vfreex — COMMENTED** [review]
/lgtm

**Comment by ashwindasr** [comment]
/hold

@MartinPl0 can you sync the changes from the art-bot PR as well?

File: Dockerfile.base [hunk 1]
```diff
@@ -56,19 +56,23 @@ USER "$USER_UID"
 WORKDIR /workspaces/art-dash
 
 USER 0
-# install dependencies (allow even openshift's random user to see)
-ENV PATH=/home/"$USERNAME"/.local/bin:/home/"$USERNAME"/bin:"$PATH"
-ENV HOME=/home/"$USERNAME"
+
+# Install Python, upgrade pip, and Git, and clone art-tools
+RUN dnf install -y git python3.8 python3-pip \
```
Comment (ashwindasr) [inline Dockerfile.base:61]: Python is already installed in the container, so this line will be redundant

File: Dockerfile.base [hunk 2]
```diff
@@ -56,19 +56,23 @@ USER "$USER_UID"
...
+ && pip3 install --upgrade pip \
```
Comment (ashwindasr) [inline Dockerfile.base:62]: Lets move this to the top of the file like art-bot's

File: Dockerfile.base [hunk 3]
```diff
@@ -56,19 +56,23 @@ USER "$USER_UID"
...
+ && git clone https://github.com/openshift-eng/art-tools.git /art-tools
+
+# Run the install.sh script from art-tools
+WORKDIR /art-tools
```
Comment (ashwindasr) [inline Dockerfile.base:66]: Lets not set a `WORKDIR` here

**Review by ashwindasr — CHANGES_REQUESTED** [review]

File: Dockerfile.update [hunk 4]
```diff
@@ -3,29 +3,31 @@ FROM art-dash-server:base
...
-USER 0
-
-# Run the install.sh script from art-tools
-WORKDIR /art-tools
-RUN git pull
-RUN ./install.sh
-
 # Return to the original work directory
-WORKDIR /workspaces/art-dash-server
+WORKDIR /workspaces/art-dash
```
Comment (ashwindasr) [inline Dockerfile.update:16]: Need to match the one in Dockerfile.base

**Review by ashwindasr — APPROVED** [review]

**Pull request merged**